src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -25,13 +64,32 @@ public static void PopulateDefaultPayload(IEnumerable<string> disabledFields, IH
                     switch (fieldName)
                     {
                         case "runtimeFramework":
-                            provider.AddHeartbeatProperty(fieldName, GetRuntimeFrameworkVer(), true);
+                            provider.AddHeartbeatProperty(fieldName, true, HeartbeatDefaultPayload.GetRuntimeFrameworkVer(), true);
```

Comment:
I will ensure there is a facility to capture any exceptions here, they will be logged but non-fatal.

Reply:
They are try...catch'ed - if you look outside my switch statement. Since it's not fatal for a heartbeat property to fail, why wouldn't we just catch the exception and move on? 
I think there is an argument here to have specific try...catch blocks per-property maybe, what do you think? My assumption when I first wrote this is that each property was (a) going to be fairly trivial to get, and (b) was going to be non-fatal should something go wrong.

Reply:
test that if there is an issue collecting one of the item, it should not affect the rest of the items. It looks to me like any issue with any item will cause the rest of the item to also be not collected.

Reply:
As discussed offline, this looks correct. Gh did not show the try..catch {}. Sorry for confusion!.
Btw - if there is no unit test covering ErrorOnOnePropertyDoesNotAffectOtherProperty, then please add one.